sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -207,62 +205,59 @@ internal string QueryAnswerRaw
 
                 if (QueryAnswer.HasValue)
                 {
-                    queryAnswerStringValue = $"{QueryAnswer.Value}{QueryAnswerCountRawSplitter}{QueryAnswerCount.GetValueOrDefault(1)}{QueryAnswerThresholdRawSplitter}{QueryAnswerThreshold.GetValueOrDefault(0.7)}";
+                    var queryAnswerValues = new List<string>();
+
+                    if (QueryAnswerCount.HasValue)
+                    {
+                        queryAnswerValues.Add($"{QueryAnswerCountRaw}{QueryAnswerCount}");
+                    }
+
+                    if (QueryAnswerThreshold.HasValue)
+                    {
+                        queryAnswerValues.Add($"{QueryAnswerThresholdRaw}{QueryAnswerThreshold}");
+                    }
+
+                    queryAnswerStringValue = $"{QueryAnswer.Value}" + (queryAnswerValues.Count > 0 ? $"|{string.Join(",", queryAnswerValues)}" : "");
                 }
 
                 return queryAnswerStringValue;
             }
-
             set
             {
-                if (string.IsNullOrEmpty(value))
-                {
-                    QueryAnswer = null;
-                    QueryAnswerCount = null;
-                    QueryAnswerThreshold = null;
-                }
-                else
+                if (!string.IsNullOrEmpty(value)) // If the value is - "extractive" or "extractive|count-1" or "extractive|threshold-0.7" or "extractive|count-5,threshold-0.9" or "extractive|threshold-0.8,count-4"
                 {
-                    if (value.Contains(QueryAnswerCountRawSplitter) || value.Contains(QueryAnswerThresholdRawSplitter))
+                    string[] queryAnswerValues = value.Split('|');
+                    if (!string.IsNullOrEmpty(queryAnswerValues[0]))
                     {
-                        string[] queryAnswerValues = value.Split('|');
-
-                        var queryAnswerPart = queryAnswerValues[0];
-                        if (string.IsNullOrEmpty(queryAnswerPart))
-                        {
-                            QueryAnswer = null;
-                        }
-                        else
-                        {
-                            QueryAnswer = new QueryAnswerType(queryAnswerPart);
-                        }
+                        QueryAnswer = new QueryAnswerType(queryAnswerValues[0]);
+                    }
 
-                        foreach (var queryAnswerValue in queryAnswerValues)
+                    if (queryAnswerValues.Length == 2)
```

Comment:
Also, can it only be of the form 
"extractive|threshold-0.7,count-5" or can the user pass 
"threshold-0.7,count-5|extractive" - the current logic assumes the first one and will break if second is a possible input.

Reply:
The first value will always be QueryAnswer.

For example:
"extractive"
"extractive|count-1,threshold-0.7"
"extractive|threshold-0.7,count-1"
"extractive|count-1"
"extractive|threshold-0.7"